File: 637-average-of-levels-in-binary-tree/637-average-of-levels-in-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def averageOfLevels(self, root: Optional[TreeNode]) -> List[float]:
        if root==None: 
            return 0
        result=[]
        self.BFS(root,result)
        return result
    def BFS(self,root:Optional[TreeNode],result:List[float])->None:
        q=[]
        q.append(root)
        while(len(q)>0):
            size=len(q)
            avg=0.0
            for i in range(0,len(q)):
                currNode=q.pop(0)
                avg+=currNode.val
                if currNode.left!=None: 
                    q.append(currNode.left)
                if currNode.right!=None: 
                    q.append(currNode.right)
            avg/=size
            result.append(avg)
    
                

File: 637-average-of-levels-in-binary-tree/637-average-of-levels-in-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    List<Double> result=new ArrayList<>();
    private void bfs(TreeNode root){
        Queue<TreeNode> q=new LinkedList<>();
        q.add(root);
        while(!q.isEmpty()){
            int size=q.size();
            double sum=0;
            for(int i=0;i<size;i++){
                TreeNode curr=q.remove();
                sum+=curr.val;
                if(curr.left!=null) q.add(curr.left);
                if(curr.right!=null) q.add(curr.right);
            }
            sum/=size;
            result.add(sum);
        }
    }
    public List<Double> averageOfLevels(TreeNode root) {
        bfs(root);
        return result;
    }
}